### PR TITLE
[cli] Add ability to connect Git provider repo

### PR DESCRIPTION
**Pull request by MatthewStanciu**
This PR adds a new subcommand `vc git`, which allows you to create a Git provider repository from your local `.git` config to your Vercel project.

Usage:

- `vc git connect` – connects a Git provider repository
- `vc git disconnect` – disconnects a Git provider repository

#### Tests

- [x] The code changed/added as part of this PR has been covered with tests
- [x] All tests pass locally with `yarn test-unit`

#### Code Review

- [ ] This PR has a concise title and thorough description useful to a reviewer
- [ ] Issue from task tracker has a link to this PR

### CODE REVIEW

File: packages/cli/src/commands/project/index.ts [hunk 1]
```diff
@@ -74,24 +77,39 @@ export default async function main(client: Client) {
...
+  let team = null;
 
-  try {
-    ({ contextName } = await getScope(client));
-  } catch (error) {
-    if (error.code === 'NOT_AUTHORIZED' || error.code === 'TEAM_DELETED') {
-      output.error(error.message);
-      return 1;
+  if (subcommand === 'connect') {
+    // || subcommand === 'disconnect'
```
Comment (EndangeredMassa) [inline packages/cli/src/commands/project/index.ts]: remove?

File: packages/cli/src/util/projects/connect-git-provider.ts [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json',
+      },
+      body: JSON.stringify({
+        type,
+        repo,
+      }),
+    })
+    .catch(err => {
```
Comment (EndangeredMassa) [inline packages/cli/src/util/projects/connect-git-provider.ts]: Can we use `try-catch` instead? I think we have a convention around preferring that over promise handling.

File: packages/cli/src/util/projects/connect-git-provider.ts [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+    case 'gitlab':
+      return 'GitLab';
+    case 'bitbucket':
+      return 'Bitbucket';
+    default:
+      return type;
+  }
+}
+
+export function parseRepoUrl(originUrl: string): {
```
Comment (EndangeredMassa) [inline packages/cli/src/util/projects/connect-git-provider.ts:83]: Can we add tests for this?

**Review by EndangeredMassa — APPROVED** [review]

**Review by EndangeredMassa — APPROVED** [review]

File: packages/cli/src/commands/git/connect.ts [hunk 4]
```diff
@@ -0,0 +1,165 @@
...
+      `No local git repo found. Run ${chalk.cyan(
+        '`git clone <url>`'
+      )} to clone a remote Git repository first.`
+    );
+    return 1;
+  }
+  const remoteUrl = pluckRemoteUrl(gitConfig);
+  if (!remoteUrl) {
+    output.error(
+      `No remote origin URL found in your Git config. Make sure you've connected your local Git repo to a Git provider first.`
```
Comment (EndangeredMassa) [inline packages/cli/src/commands/git/connect.ts]:
```suggestion
      `No remote origin URL found in your Git config. Make sure you've configured a remote repo in your local git config. Run ${chalk.cyan('`git remote --help`')} for more details.`
```

I think using the word "connected" here might be confusing in an error message after the user tried to `vc git connect`. What do you think about this?

File: packages/cli/src/commands/git/connect.ts [hunk 5]
```diff
@@ -0,0 +1,165 @@
...
+  }
+  const parsedUrl = parseRepoUrl(remoteUrl);
+  if (!parsedUrl) {
+    output.error(
+      `Failed to parse Git repo data from the following remote URL in your Git config: ${link(
+        remoteUrl
+      )}`
+    );
+    return 1;
+  }
```
Comment (EndangeredMassa) [inline packages/cli/src/commands/git/connect.ts:85]: Should we log that we identified the git remote here? When testing this out, I imagine a user being a bit confused about where the git remote came from.

What I see:

```
$  vc git connect
Vercel CLI 27.0.2-495866b
> Connected GitHub repository EndangeredMassa/nuxt-v3!
```

What I'd like to see:

```
$  vc git connect
Vercel CLI 27.0.2-495866b
> Identified git remote "origin": https://github.com/EndangeredMassa/nuxt-v3.git
> Connected GitHub repository EndangeredMassa/nuxt-v3!
```

File: packages/cli/src/commands/git/index.ts [hunk 6]
```diff
@@ -0,0 +1,94 @@
...
+import connect from './connect';
+import disconnect from './disconnect';
+
+const help = () => {
+  console.log(`
+  ${chalk.bold(`${logo} ${getPkgName()} git`)} <command>
+
+  ${chalk.dim('Commands:')}
+
+    connect                          Connect a Git provider repository to your project
```
Comment (EndangeredMassa) [inline packages/cli/src/commands/git/index.ts]:
```suggestion
    connect                          Connect your git config "origin" remote as a Git provider to your project
```

Can we have this say where the remote repo comes from?

File: packages/cli/src/commands/git/index.ts [hunk 7]
```diff
@@ -0,0 +1,94 @@
...
+    disconnect                       Disconnect a Git provider repository from your project
```
Comment (EndangeredMassa) [inline packages/cli/src/commands/git/index.ts]:
```suggestion
    disconnect                       Disconnect the Git provider repository from your project
```

File: packages/cli/src/commands/git/index.ts [hunk 8]
```diff
@@ -0,0 +1,94 @@
...
+    'TOKEN'
+  )}        Login token
+
+  ${chalk.dim('Examples:')}
+
+  ${chalk.gray('–')} Connect a Git provider repository
+
+    ${chalk.cyan(`$ ${getPkgName()} git connect`)}
+
+  ${chalk.gray('–')} Disconnect a Git provider repository
```
Comment (EndangeredMassa) [inline packages/cli/src/commands/git/index.ts]:
```suggestion
  ${chalk.gray('–')} Disconnect the Git provider repository
```

File: packages/cli/src/commands/deploy/args.ts [hunk 9]
```diff
@@ -15,6 +15,7 @@ export const help = () => `
       )}
       dev                              Start a local development server
       env                              Manages the Environment Variables for your current Project
+      git                              Connects a Git provider repository to your project from your local Git config
```
Comment (EndangeredMassa) [inline packages/cli/src/commands/deploy/args.ts]:
```suggestion
      git                              Manage Git provider repository for your current Project
```

**Review by EndangeredMassa — APPROVED** [review]

**Comment by MatthewStanciu** [comment]
Thanks @EndangeredMassa 🙌  feel much happier with the copy after these changes

File: packages/cli/src/commands/git/connect.ts (hunk 5 above)
Comment (unitof) [reply packages/cli/src/commands/git/connect.ts:85]: Indeed I was a bit confused (pleasantly!). I often rename my `origin` to `github` but `vc git connect` still picked it up!

**Pull request merged**